x-pack/plugins/ingest_pipelines/server/routes/api/get.ts

```diff
@@ -18,33 +18,22 @@ const paramsSchema = schema.object({
 export const registerGetRoutes = ({
   router,
   license,
-  lib: { isEsError },
+  lib: { handleEsError },
 }: RouteDependencies): void => {
   // Get all pipelines
   router.get(
     { path: API_BASE_PATH, validate: false },
     license.guardApiRoute(async (ctx, req, res) => {
-      const { callAsCurrentUser } = ctx.core.elasticsearch.legacy.client;
+      const { client: clusterClient } = ctx.core.elasticsearch;
 
       try {
-        const pipelines = await callAsCurrentUser('ingest.getPipeline');
+        const { body: pipelines } = await clusterClient.asCurrentUser.ingest.getPipeline({
+          id: '_all',
+        });
 
         return res.ok({ body: deserializePipelines(pipelines) });
       } catch (error) {
-        if (isEsError(error)) {
-          // ES returns 404 when there are no pipelines
-          // Instead, we return an empty array and 200 status back to the client
-          if (error.status === 404) {
```

Comment:
Is this check no longer needed?

Reply:
Good point! This behaviour needs to be preserved!